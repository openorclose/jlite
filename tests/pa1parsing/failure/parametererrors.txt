class ParameterErrors {
  Void main() {
    return 1;
  }
}

class Param {
  Void name(noType, NoId, random stuff here, more random if while random here) {
  // it can detect missing Type, missing Id, and partially help if you put in wrong things there
      readln(a);
  }

  Void (noType) { //it can detect a missing identifier
    return;
  }

  Class (noId, noId, noType) {
        return;
  }

  Class a () {
      return;
    }

    name(noType) { return; } // missing Class it can't do so well though. on trying to detect this error it consumes the previous input.
}