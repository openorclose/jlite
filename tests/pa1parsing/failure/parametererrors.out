Error in line 8, column 34 : Syntax error
instead expected token classes are []
On line 8, column 27:
Error : Your parameter definition should be of the form <Type> <identifier>.

On line 8, column 21:
Error : Your parameter with type `NoId` is missing its name.

On line 8, column 13:
Error : Your parameter named `noType` is missing its type.

Error in line 13, column 8 : Syntax error
instead expected token classes are [ID]
On line 13, column 9:
Error : Your parameter named `noType` is missing its type.

On line 13, column 15:
Error : Your method declaration has to have a type and an identifier.

Error in line 17, column 9 : Syntax error
instead expected token classes are [ID]
On line 17, column 22:
Error : Your parameter named `noType` is missing its type.

On line 17, column 16:
Error : Your parameter named `noId` is missing its type.

On line 17, column 10:
Error : Your parameter named `noId` is missing its type.

On line 17, column 28:
Error : Your method declaration has to have a type and an identifier.

Error in line 25, column 5 : Syntax error
instead expected token classes are [error, INT, BOOL, STRING, VOID, RBRACE, CNAME]
On line 21, column 14:
Error : Method body cannot be parsed.
Check that there is at least one non-variable-declaration statement, and that all variable declaration statements come before other statements.

Error : 10 syntax errors detected. Parse failed.
